src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx

```diff
@@ -0,0 +1,201 @@
+      name: Label.CANCEL_BUTTON,
+    });
+    expect(cancelButton).toBeInTheDocument();
+    expect(
+      screen.getByRole("button", { name: Label.CONFIRM_BUTTON }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText("APPLICATION COUNT (UNIT COUNT)"),
+    ).toBeInTheDocument();
+    expect(screen.getByText("2 (4)")).toBeInTheDocument();
+    await userEvent.click(cancelButton);
+    expect(mockSetConfirmType).toHaveBeenCalledWith(null);
+  });
+
+  it("should run selected action and remove panel query params", async () => {
+    const executeActionOnUnitsSpy = vi
+      .fn()
+      .mockImplementation(() => Promise.resolve());
+    vi.spyOn(actionsHooks, "useExecuteActionOnUnits").mockImplementation(
+      () => executeActionOnUnitsSpy,
+    );
+    renderComponent(
+      <ConfirmationDialog
+        confirmType={ConfirmType.SUBMIT}
+        selectedAction={"add-disk"}
+        selectedApplications={mockSelectedApplcations}
+        setConfirmType={mockSetConfirmType}
+        selectedActionOptionValue={{
+          bucket: "",
+          "osd-devices": "new device",
+        }}
+        onRemovePanelQueryParams={mockOnRemovePanelQueryParams}
+      />,
+      { state, path, url },
+    );
+    await userEvent.click(
+      screen.getByRole("button", { name: Label.CONFIRM_BUTTON }),
+    );
+    expect(mockSetConfirmType).toHaveBeenCalledWith(null);
+    const call = executeActionOnUnitsSpy.mock.calls[0];
+    expect(call[1]).toBe("add-disk");
+    expect(call[0]).toEqual(["ceph-0", "ceph-1"]);
+    expect(call[2]).toEqual({
+      bucket: "",
+      "osd-devices": "new device",
+    });
+    expect(await screen.findByText(Label.ACTION_SUCCESS)).toBeInTheDocument();
+    expect(mockOnRemovePanelQueryParams).toHaveBeenCalledOnce();
+  });
+
+  it("should display error toast if action fails", async () => {
+    const executeActionOnUnitsSpy = vi
+      .fn()
+      .mockImplementation(() => Promise.reject(new Error()));
+    vi.spyOn(actionsHooks, "useExecuteActionOnUnits").mockImplementation(
+      () => executeActionOnUnitsSpy,
+    );
+    renderComponent(
+      <ConfirmationDialog
+        confirmType={ConfirmType.SUBMIT}
+        selectedAction={"add-disk"}
+        selectedApplications={mockSelectedApplcations}
+        setConfirmType={mockSetConfirmType}
+        selectedActionOptionValue={{}}
+        onRemovePanelQueryParams={mockOnRemovePanelQueryParams}
+      />,
+      { state, path, url },
+    );
+    await userEvent.click(
+      screen.getByRole("button", { name: Label.CONFIRM_BUTTON }),
+    );
+    expect(mockSetConfirmType).toHaveBeenCalledWith(null);
+    const call = executeActionOnUnitsSpy.mock.calls[0];
+    expect(call[1]).toBe("add-disk");
+    expect(call[0]).toEqual(["ceph-0", "ceph-1"]);
+    expect(call[2]).toEqual({});
+    expect(await screen.findByText(Label.ACTION_ERROR)).toBeInTheDocument();
+    expect(mockOnRemovePanelQueryParams).toHaveBeenCalledOnce();
```

Comment:
This test should just test that part that we're concerned with (the other lines are captured by the success case above).
```suggestion
    expect(await screen.findByText(Label.ACTION_ERROR)).toBeInTheDocument();
```